.github/workflows/preview_ci.yml

```diff
@@ -1,69 +1,76 @@
 name: Preview - CI
 on:
-  push:
-    paths-ignore:
-      - '**/*.md'
-      - 'mkdocs.yml'
-      - 'src/docs/**/*'
-    branches: [ main ]
+  workflow_dispatch:
+  schedule:
+    # 4:19 AM UTC every day. A random time to avoid peak times of GitHub Actions.
+    - cron: '19 4 * * *'
 env:
   DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
   DOTNET_CLI_TELEMETRY_OPTOUT: true
 jobs:
   test:
-    runs-on: ${{ matrix.os }}
+    runs-on: ubuntu-latest
```

Comment:
The [`main` version](https://github.com/OrchardCMS/OrchardCore/blob/a3a2b07cdd1e89bd8e570c3b944799734375eee0/.github/workflows/preview_ci.yml) has a lot of `if: matrix.os == 'ubuntu-latest'` checks, because everything related to NuGet and Docker publishing actually only happened under Ubuntu. Now, we have two worfklows:

- A build and test one for every `main` commit that runs both Ubuntu and Windows.
- A preview publishing one that runs once a day and only under Ubuntu that does Cloudsmith and Docker publishing. Since everything this does requires Ubuntu, it only runs under Ubuntu.

Reply:
I am not sure what is the benefit of running the test after the code is pushed to main. At that point, it should have been already tested and does not need additional checks. I think all the testing should be done in the PR "which we already do". In the preview, we probably do not need testing since no code will make it to main unless it goes through the PR pipeline which does the testing on Ubuntu and Windows.

Reply:
It's not impossible to push to `main` directly, just allowed for fewer people. I assume this is why this workflow has run tests for all `main` commits. Note that that part I didn't add, just kept.

Reply:
No one can push directly to main, unless they remove the option in the project settings. But I don't think we need to worry about it. IMO, main_ci can be removed completely with the new logic. PR CI ensures the new code will work on windows and Ubuntu. the preview CI will run the project and deploy a docker image and the nuget packages.

Reply:
I can push directly to main, just don't tell anyone ;). Administrators can bypass that branch protection rule without changing the config, just as they can merge PRs without waiting for the checks to finish. Now they wouldn't, barring exceptional circumstances, of course, but they can.

But in practice, we should almost always be good with just PR builds, and not waste computing resources with double-builds.

@sebastienros do you have some insights from Microsoft projects, perhaps?

BTW, and this will be super subjective, but I like this green checkmark:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/0b26a080-9f2e-4394-b8a1-85b622ba28a3)

When checking out a project on GitHub, this adds a bit of confidence, that the project does some automated checks. Now, this would actually remain due to Validate Pull Request, just without any actual build.

Reply:
I also think running tests on a system is important. Maybe we can merge a branch that is not totally rebased on main and would break something. Linux tests are fast enough.

Reply:
Ah good point, we can actually merge PRs that aren't in sync with the latest `main` (we had a discussion about not enforcing that because people were complaining). So, running the workflow on `main` guards against issues coming from that.